packages/openneuro-components/src/front-page/GetUpdates.tsx

```diff
@@ -2,23 +2,95 @@ import React from 'react'
 import { Button } from '../button/Button'
 import { Input } from '../input/Input'
 
-export const GetUpdates = () => (
-  <div className="get-updates-bar">
-    <div className="get-updates-label">
-      <h2>Get Updates</h2>
-      <h3>Find out about new version and future releases.</h3>
-    </div>
-    <div className="get-updates-form">
-      <Input
-        type="email"
-        label="Email Address"
-        placeholder="email@openneuro.com"
-        labelStyle="float"
-        name="float-example"
-        value=""
-        setValue={() => {}}
-      />
-      <Button secondary label="Subscribe" />
+export const modes = {
+  // display prompt / email input / submit button
+  GET: Symbol('get'),
+  // display success message / done
+  SUCCESS: Symbol('success'),
+  // display error message / try again button
+  ERROR: Symbol('error'),
+}
+
+export const submitHandler = (subscribe, mode, setMode, value) => e => {
+  e.preventDefault()
+  switch (mode) {
+    case modes.SUCCESS:
+      break
+    case modes.ERROR:
+      setMode(modes.GET)
+      break
+    case modes.GET:
+    default:
+      subscribe(value, result => {
+        if (result.data) {
+          setMode(
+            result.data.subscribeToNewsletter ? modes.SUCCESS : modes.ERROR,
+          )
+        }
+      })
+      break
+  }
+}
+
+export const getText = mode => {
+  switch (mode) {
+    case modes.ERROR:
+      return {
+        headingText: 'Whoops.',
+        messageText:
+          'We were unable to sign you up for updates. Please try again.',
+        buttonText: 'Try Again',
+      }
+    case modes.SUCCESS:
+      return {
+        headingText: 'Success!',
+        messageText: 'You are now signed up to receive news updates.',
+        buttonText: 'Done',
+      }
+    case modes.GET:
+    default:
+      return {
+        headingText: 'Get Updates',
+        buttonText: 'Subscribe',
+        messageText: 'Find out about new version and future releases',
+      }
+  }
+}
+
+export const GetUpdates = ({ subscribe, initialMode = modes.GET }) => {
+  const [mode, setMode] = React.useState(initialMode)
+  const [value, setValue] = React.useState('')
+  const { headingText, messageText, buttonText } = getText(mode)
+  return (
+    <div className="get-updates-bar">
+      <div className="get-updates-label">
+        <h2>{headingText}</h2>
+        <h3>{messageText}</h3>
+      </div>
+      <form
+        onSubmit={submitHandler(subscribe, mode, setMode, value)}
+        className="get-updates-form">
+        <>
+          <div className="">{}</div>
+          {mode === modes.ERROR || mode === modes.SUCCESS ? null : (
```

Comment:
```suggestion
          {mode === modes.GET && (
```
I think this would be more readable.